aten/src/ATen/OpaqueTensorImpl.h

```diff
@@ -99,6 +99,16 @@ c10::intrusive_ptr<TensorImpl> shallow_copy_and_detach() const override {
   impl->is_contiguous_ = is_contiguous_;
   impl->is_wrapped_number_ = is_wrapped_number_;
   impl->reserved_ = reserved_;
+  switch (version_counter_mode) {
```

Comment:
I'm not quite sure how this fits into your other changes, which IIRC gets rid of some of this duplicate code.  Will this code still be duplicated in all of the derived classes once all the outstanding PRs are merged?

Reply:
This code won't be duplicated in any of the derived classes after merging all of the outstanding PRs - it will be moved into `copy_tensor_data()` in https://github.com/pytorch/pytorch/pull/17072, which is called by all derived classes of TensorImpl to copy the base TensorImpl's fields.